Stratis.Bitcoin.Features.Miner/PosBlockAssembler.cs

```diff
@@ -53,7 +53,7 @@ protected override void UpdateHeaders()
             base.UpdateHeaders();
 
             var stake = new BlockStake(this.pblock);
-            this.pblock.Header.Bits = StakeValidator.GetNextTargetRequired(this.stakeChain, this.chain.Tip, this.network.Consensus, this.options.IsProofOfStake);
+            this.pblock.Header.Bits = StakeValidator.GetNextTargetRequired(this.stakeChain, this.consensusLoop.Tip, this.network.Consensus, this.options.IsProofOfStake);
```

Comment:
I see why you did this be in reality we should never be staking if TIP are not the same.
In fact this might be buggy as if tips are not the same we are staking on the wrong height and will create a split.
The solution is to throw and a miner exception and let the processes start again.

Reply:
just run into a bug which crashed my miner - see #500 
using of chain.Tip is to be very careful, consensus is better, but still not good really enough
all these tips should come from above (parameter)

chain.Tip can change at any time, if we at any point rely on it, it is an open vector for a bug, but this should be solved/discussed in #500

Reply:
Also consensus tip can change in flight I agree we need to create a param.
Maybe better take a param and compare tips else throw?

Reply:
I think argument suggestion as in #500 will be just good enough. You don't need to check it really. You check once when you start the round, then you use the value for that round, that should be enough. So either we don't mine it, then next round we check again, or we mine it, in which case, if we fix #488, it will at that point recognize that we were mining something else (and there we have to decide whether we want to continue with our own or jump on remote train, better strategy is probably to continue with our own).